src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -46,6 +55,26 @@ def make_install_req_from_link(link, parent):
     )
 
 
+def make_install_req_from_editable(link, parent):
+    # type: (Link, InstallRequirement) -> InstallRequirement
+    assert parent.editable, "parent not editable"
+    return install_req_from_editable(
+        link.url,
+        # HACK: install_req_from_editable accepts Optional[str] here, but
+        # parent.comes_from is Union[str, InstallRequirement, None]. How do
+        # we fix the type hint conflicts?
+        comes_from=parent.comes_from,  # type: ignore
+        use_pep517=parent.use_pep517,
+        isolated=parent.isolated,
+        constraint=parent.constraint,
+        options=dict(
+            install_options=parent.install_options,
+            global_options=parent.global_options,
+            hashes=parent.hash_options
+        ),
+    )
+
+
```

Comment:
There's another "do we need to support editables?" TODO just below this, that I guess can go now. Or do you think there's something still needed for installed candidates where we do different things depending on whether they are editable installs?

Reply:
I simply forgot to remove it, but that’s a good question now you mention it… is there a difference?

Reply:
I simply forgot to remove it, but now that you mention it, that’s a good question… @pradyunsg do you know whether there is a difference?

Reply:
I don't think there is a difference, so it should be find just to remove the comment. I only asked because I think it was you who added the comment in the first place, so you may have had something specific in mind.

Reply:
I’ve removed the comment. We will always know if this comes up as a bug :p